libyul/AsmAnalysis.cpp

```diff
@@ -417,24 +417,31 @@ std::vector<YulString> AsmAnalyzer::operator()(FunctionCall const& _funCall)
 				std::string functionName = _funCall.functionName.name.str();
 				if (functionName == "datasize" || functionName == "dataoffset")
 				{
-					if (!m_dataNames.count(std::get<Literal>(arg).value))
+					auto const& argumentAsLiteral = std::get<Literal>(arg);
+					if (!m_dataNames.count(YulString(formatLiteral(argumentAsLiteral))))
 						m_errorReporter.typeError(
 							3517_error,
 							nativeLocationOf(arg),
-							"Unknown data object \"" + std::get<Literal>(arg).value.str() + "\"."
+							"Unknown data object \"" + formatLiteral(argumentAsLiteral) + "\"."
 						);
 				}
 				else if (functionName.substr(0, "verbatim_"s.size()) == "verbatim_")
 				{
-					if (std::get<Literal>(arg).value.empty())
+					static u256 const empty {valueOfStringLiteral("").value()};
+					auto const& literalValue = std::get<Literal>(arg).value;
+					if ((literalValue.unlimited() && literalValue.builtinStringLiteralValue().empty()) || (!literalValue.unlimited() && literalValue.value() == empty))
 						m_errorReporter.typeError(
 							1844_error,
 							nativeLocationOf(arg),
 							"The \"verbatim_*\" builtins cannot be used with empty bytecode."
 						);
 				}
-
-				argTypes.emplace_back(expectUnlimitedStringLiteral(std::get<Literal>(arg)));
+				bool const isUnlimitedLiteralArgument = [&]() {
+					if (BuiltinFunction const* f = m_dialect.builtin(_funCall.functionName.name))
+						return f->literalArguments.size() >= i && f->literalArguments.at(i-1).has_value();
+					return false;
+				}();
+				argTypes.emplace_back(expectStringLiteral(std::get<Literal>(arg), isUnlimitedLiteralArgument));
 				continue;
```

Comment:
BTW, `isUnlimitedLiteralArgument` is redundant anyway. It will always be true here because we're inside this overcomplicated condition that already checked that this is the case:

```c++
		if (
			auto literalArgumentKind = (literalArguments && i <= literalArguments->size()) ?
				literalArguments->at(i - 1) :
				std::nullopt
		)
```

Reply:
nice, done that:)